extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/iam/IamServiceImpl.java

```diff
@@ -172,18 +209,40 @@ private Supplier<IamCredentialsClient> defaultIamCredentialsClientSupplier() {
         }
     }
 
-    private record ApplicationDefaultCredentials(Monitor monitor) implements AccessTokenProvider {
+    private record DefaultCredentialsUtil(Monitor monitor) implements CredentialsUtil {
         @Override
-        public GcpAccessToken getAccessToken() {
+        public GoogleCredentials getApplicationDefaultCredentials() {
+            try {
+                return GoogleCredentials.getApplicationDefault();
+            } catch (IOException ioException) {
+                monitor.severe("Cannot get application default credentials", ioException);
+                throw new GcpException(ioException);
+            }
+        }
+
+        @Override
+        public void refreshCredentials(GoogleCredentials credentials) {
             try {
-                var credentials = GoogleCredentials.getApplicationDefault().createScoped(IamScopes.CLOUD_PLATFORM);
                 credentials.refreshIfExpired();
-                var token = credentials.getAccessToken();
-                return new GcpAccessToken(token.getTokenValue(), token.getExpirationTime().getTime());
             } catch (IOException ioException) {
-                monitor.severe("Cannot get application default access token", ioException);
-                return null;
+                monitor.severe("Cannot get refresh the credentials", ioException);
+                throw new GcpException(ioException);
             }
         }
+
+        @Override
+        public GoogleCredentials createScoped(GoogleCredentials credentials, String... scopes) {
+            return credentials.createScoped(scopes);
+        }
```

Comment:
this method doesn't look like it needs to stay here, as the behavior for create scoped credentials is already provided by the `GoogleCredentials` class and it's pretty straightforward

Reply:
Done